client/app/components/dashboards/dashboard-widget/VisualizationWidget.jsx

```diff
@@ -94,9 +94,11 @@ function VisualizationWidgetHeader({ widget, refreshStartedAt, parameters, onPar
           <p>
             <QueryLink query={widget.getQuery()} visualization={widget.visualization} readOnly={!canViewQuery} />
           </p>
-          <HtmlContent className="text-muted markdown query--description">
-            {markdown.toHTML(widget.getQuery().description || "")}
-          </HtmlContent>
+          {!isEmpty(widget.getQuery().description) && (
+            <HtmlContent className="text-muted markdown query--description">
+              {markdown.toHTML(widget.getQuery().description || "")}
```

Comment:
How about `markdown.toHTML` performance? Is it fast enough, or maybe also needs some caching?

Reply:
I didn't cache it with `useMemo` because it runs `widget.getQuery()` currently, it would need to have the `widget` as a dependency. Since that component is basically re-rendered when `widget`changes, that didn't seem that relevant :thinking:. Also now it only runs that function when the query has a description.